Source/WebCore/page/Navigation.cpp

```diff
@@ -168,7 +168,7 @@ void Navigation::initializeForNewWindow(std::optional<NavigationNavigationType>
 
     // https://html.spec.whatwg.org/multipage/browsing-the-web.html#getting-session-history-entries-for-the-navigation-api
     Vector<Ref<HistoryItem>> items;
-    auto rawEntries = page->checkedBackForward()->allItems();
+    auto rawEntries = frame()->isMainFrame() ? page->backForward().allItems() : page->backForward().itemsForFrame(frame()->frameID());
```

Comment:
Is it correct for the main frame to use all items including ones from its subframes? Is there a test exercising that?

Reply:
Oh, good question. I will read the spec carefully. My initial feeling is you are right. It must be restricted to main frame items. https://html.spec.whatwg.org/multipage/browsing-the-web.html#getting-session-history-entries